backend/ddinter_loader.py

```
@@ -0,0 +1,381 @@
+    DDInter drops/renames a pair between releases."""
+    if cache_dir is not None:
+        global DDINTER_CACHE_DIR
+        DDINTER_CACHE_DIR = cache_dir
+
+    paths = [_fetch_csv(c, force_refresh=force_refresh) for c in DDINTER_ATC_CLASSES]
+
+    conn = sqlite3.connect(DB_PATH)
+    conn.execute("PRAGMA busy_timeout = 5000")
+    try:
+        ensure_interactions_table(conn)
+        conn.execute(f"DELETE FROM {TABLE_NAME}")
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)